library/std/src/sys/unix/thread_parking/netbsd.rs

```diff
@@ -0,0 +1,54 @@
+#![cfg(target_os = "netbsd")]
+
+use crate::ffi::{c_int, c_void};
+use crate::ptr::{null, null_mut};
+use crate::time::Duration;
+use libc::{_lwp_self, clockid_t, lwpid_t, time_t, timespec, CLOCK_MONOTONIC};
+
+extern "C" {
+    fn ___lwp_park60(
+        clock_id: clockid_t,
+        flags: c_int,
+        ts: *mut timespec,
+        unpark: lwpid_t,
+        hint: *const c_void,
+        unparkhint: *const c_void,
+    ) -> c_int;
+    fn _lwp_unpark(lwp: lwpid_t, hint: *const c_void) -> c_int;
+}
+
+pub type ThreadId = lwpid_t;
+
+#[inline]
+pub fn current() -> ThreadId {
+    unsafe { _lwp_self() }
+}
+
+#[inline]
+pub fn park() {
+    unsafe {
+        ___lwp_park60(0, 0, null_mut(), 0, null(), null());
```

Comment:
What happened to the `hint`?

Reply:
Since it does not exist on SGX and SOLID, I did not include it here. I will add it in again, since it can be optimized away easily when not used.